### ISSUE TITLE
Problems with column types for distribution control

### ISSUE THREAD

**Issue by dannyparsons** [africanmathsinitiative/R-Instat#6641]
- `ucrDistribution.strDataType` is sometimes not set correctly when the column has missing values.
- There is a mismatch between the possible values of `ucrDistribution.strDataType` and the values expected on some dialogs like `dlgFitModel.strVariableType`
- The `integer` value returned by the function isn't recognised by `ucrDistribution`

Particularly noticed with integer/positive integer columns with missing values before non-missing values.

**Comment by dannyparsons** [comment]
@Patowhiz The PR #6642 fixes the immediate bugs here but a better solution would be to make `strDistributionType` an `Enum` so that we specify its possible values in one place and get easily keep this in sync with the R function. Can you work on this tidying up once the PR is merged?

**Comment by Patowhiz** [comment]
@dannyparsons yes I can tidy this up.

**Comment by dannyparsons** [comment]
@Patowhiz The first PR has been merged now. Could you open a new issue for the tidying you will do? This isn't urgent so not needed right away.
